tests/qat/test_qasm.py

```diff
@@ -228,6 +228,46 @@ def test_cx_override(self):
             == 2
         )
 
+    def test_ecr_override(self):
+        hw = get_default_echo_hardware(4)
+        parser = Qasm3Parser()
+        result = parser.parse(get_builder(hw), get_qasm3("ecr_override_test.qasm"))
+        # tests that there are 2 extra_soft_square pulses from the custom def
+        assert (
+            len(
+                [
+                    inst
+                    for inst in result.instructions
+                    if hasattr(inst, "shape")
+                    and (inst.shape is PulseShapeType.EXTRA_SOFT_SQUARE)
+                ]
+            )
```

Comment:
not robust enough, test should test parsing result is it matches what's in the qasm, and then we can add assertions on the type of the generated instructions for example, also test the targets of the instruction: expect a Pulse to target logical pulse channel (drive for example) associated to the qubits in parameters ... lots of things you can check